scripts/lib/libvirt/test_libvirt_setup.py

```diff
@@ -80,21 +80,30 @@ def test_net_config(self):
 
 class TestLibvirtAdminConfig(unittest.TestCase):
 
-    def test_admin_config(self):
-        args = Arguments()
-        args.cloud = "cloud"
-        args.adminnodememory = 2097152
-        args.adminvcpus = 1
-        args.emulator = "/usr/bin/qemu-system-x86_64"
-        args.adminnodedisk = "/dev/cloud/cloud.admin"
-        args.localreposrc = ""
-        args.localrepotgt = ""
-        cpu_flags = libvirt_setup.readfile(
+    def setUp(self):
+        self.args = Arguments()
+        self.args.cloud = "cloud"
+        self.args.adminnodememory = 2097152
+        self.args.adminvcpus = 1
+        self.args.emulator = "/usr/bin/qemu-system-x86_64"
+        self.args.adminnodedisk = "/dev/cloud/cloud.admin"
+        self.args.firmwaretype = ""
+        self.args.localreposrc = ""
+        self.args.localrepotgt = ""
+        self.cpu_flags = libvirt_setup.readfile(
             "{0}/cpu-intel.xml".format(TEMPLATE_DIR))
 
+    def test_admin_config(self):
         should_config = libvirt_setup.readfile(
             "{0}/cloud-admin.xml".format(FIXTURE_DIR))
-        is_config = libvirt_setup.admin_config(args, cpu_flags)
+        is_config = libvirt_setup.admin_config(self.args, self.cpu_flags)
+        self.assertEqual(is_config, should_config)
+
+    def test_admin_config_uefi(self):
+        self.args.firmwaretype = "uefi"
+        should_config = libvirt_setup.readfile(
+            "{0}/cloud-admin-uefi.xml".format(FIXTURE_DIR))
```

Comment:
this also needs some normalization, use % or {}, but if we are not using python 2.6 I would not use {0}